app/src/main/java/org/oppia/app/player/state/itemviewmodel/StateButtonViewModel.kt

```diff
@@ -0,0 +1,117 @@
+package org.oppia.app.player.state.itemviewmodel
+
+import android.content.Context
+import android.widget.Button
+import androidx.databinding.BindingAdapter
+import androidx.databinding.ObservableField
+import androidx.lifecycle.ViewModel
+import org.oppia.app.fragment.FragmentScope
+import javax.inject.Inject
+import org.oppia.app.R
+import org.oppia.app.viewmodel.ObservableViewModel
+
+private const val CONTINUE = "Continue"
+private const val END_EXPLORATION = "EndExploration"
+private const val LEARN_AGAIN = "LearnAgain"
+private const val MULTIPLE_CHOICE_INPUT = "MultipleChoiceInput"
+private const val ITEM_SELECT_INPUT = "ItemSelectionInput"
+private const val TEXT_INPUT = "TextInput"
+private const val FRACTION_INPUT = "FractionInput"
+private const val NUMERIC_INPUT = "NumericInput"
+private const val NUMERIC_WITH_UNITS = "NumberWithUnits"
+
+/** [ViewModel] for state-fragment. */
+@FragmentScope
+class StateButtonViewModel @Inject constructor(val context: Context) : ObservableViewModel() {
+  companion object {
+    @JvmStatic
+    @BindingAdapter("buttonDrawable")
+    fun setBackgroundResource(button: Button, resource: Int) {
+      button.setBackgroundResource(resource)
+    }
+  }
+
+  var isAudioFragmentVisible = ObservableField<Boolean>(false)
+
+  var isNextButtonVisible = ObservableField<Boolean>(false)
+  var isPreviousButtonVisible = ObservableField<Boolean>(false)
+
+  var observableInteractionId = ObservableField<String>()
+  var isInteractionButtonActive = ObservableField<Boolean>(false)
+  var isInteractionButtonVisible = ObservableField<Boolean>(false)
+  var drawableResourceValue = ObservableField<Int>(R.drawable.state_button_primary_background)
+
+  var name = ObservableField<String>()
+
+  fun setObservableInteractionId(interactionId: String) {
+    setNextButtonVisible(false)
+    observableInteractionId.set(interactionId)
+    when (interactionId) {
```

Comment:
Please add a TODO here on #249 to generalize this binding to make adding future interactions easier. We'll solve this post-prototype.

Reply:
Done. Added TODO